api/base/views.py

```diff
@@ -75,3 +79,21 @@ def root(request, format=None):
             'users': absolute_reverse('users:user-list'),
         }
     })
+
+
+class OsfAPIViewMeta(type):
+    """Metaclass to ensure a base level of permissions are applied to all of our views
+    """
+    def __new__(cls, name, bases, attributes):
+        permission_classes = set((
+            drf_permissions.IsAuthenticatedOrReadOnly,
+            base_permissions.TokenHasScope,
+        ))
+        bases_permission_classes = attributes.get('permission_classes', tuple())
+        try:
+            iter(bases_permission_classes)
+        except TypeError:
+            bases_permission_classes = (bases_permission_classes, )
+        bases_permission_classes = set(bases_permission_classes)
```

Comment:
Will inheritance work correctly with this? It doesn't seem like this handles inherited `permissions_classes`. It might make sense to implement `__init__` instead of `__new__`; that way, we let Python handle inheritance rather than doing it ourselves.